Corona-Warn-App/src/test/java/de/rki/coronawarnapp/datadonation/analytics/AnalyticsTest.kt

```diff
@@ -0,0 +1,191 @@
+        every { settings.analyticsEnabled } returns mockFlowPreference(false)
+        every { analyticsConfig.probabilityToSubmit } returns 1.0
+        every { settings.lastSubmittedTimestamp } returns mockFlowPreference(Instant.now())
+
+        val analytics = createInstance()
+
+        runBlockingTest2 {
+            analytics.submitIfWanted()
+        }
+
+        coVerify(exactly = 0) {
+            analytics.submitAnalyticsData()
+        }
+    }
+
+    @Test
+    fun `abort due to time since onboarding`() {
+        every { settings.analyticsEnabled } returns mockFlowPreference(false)
+        every { analyticsConfig.probabilityToSubmit } returns 1.0
+        every { settings.lastSubmittedTimestamp } returns mockFlowPreference(
+            Instant.now().minus(Days.TWO.toStandardDuration())
+        )
+        every { LocalData.onboardingCompletedTimestamp() } returns Instant.now().millis
+
+        val analytics = createInstance()
+
+        runBlockingTest2 {
+            analytics.submitIfWanted()
+        }
+
+        coVerify(exactly = 0) {
+            analytics.submitAnalyticsData()
+        }
+    }
+
+    @Test
+    fun `submit analytics data`() {
+        every { settings.analyticsEnabled } returns mockFlowPreference(true)
+        every { analyticsConfig.probabilityToSubmit } returns 1.0
+
+        val twoDaysAgo = Instant.now().minus(Days.TWO.toStandardDuration())
+        every { settings.lastSubmittedTimestamp } returns mockFlowPreference(twoDaysAgo)
+        every { LocalData.onboardingCompletedTimestamp() } returns twoDaysAgo.millis
+
+        val metadata = PpaData.ExposureRiskMetadata.newBuilder()
+            .setRiskLevel(PpaData.PPARiskLevel.RISK_LEVEL_HIGH)
+            .setMostRecentDateAtRiskLevel(java.time.Instant.ofEpochSecond(101010).toEpochMilli())
+            .setDateChangedComparedToPreviousSubmission(true)
+            .setRiskLevelChangedComparedToPreviousSubmission(true)
+            .build()
+
+        coEvery { exposureRiskMetadataDonor.beginDonation(any()) } returns
+            ExposureRiskMetadataDonor.ExposureRiskMetadataContribution(
+                contributionProto = metadata,
+                onContributionFinished = {}
+            )
+
+        coEvery { deviceAttestation.attest(any()) } returns
+            object : DeviceAttestation.Result {
+                override val accessControlProtoBuf: PpacAndroid.PPACAndroid
+                    get() = PpacAndroid.PPACAndroid.getDefaultInstance()
+
+                override fun requirePass(requirements: SafetyNetRequirements) {}
+            }
+
+        coEvery { dataDonationAnalyticsServer.uploadAnalyticsData(any()) } just Runs
+
+        val analytics = createInstance()
+
+        runBlockingTest2 {
+            analytics.submitIfWanted()
+        }
+
+        coVerify(exactly = 1) {
+            analytics.submitAnalyticsData()
+            dataDonationAnalyticsServer.uploadAnalyticsData(any())
```

Comment:
Do we have access to the data that would be uploaded such that we can be specific than `any()`?

e.g.

Can we construct `PpaDataRequestAndroid` with the same value? Or would the ProtoBuf equals/hash still not be equal and allow this?